grails-webhooks/src/test/groovy/webhooks/WebhookControllerSpec.groovy

```diff
@@ -61,6 +61,40 @@ class WebhookControllerSpec extends Specification implements ControllerUnitTest<
         "1234#test" | '{"msg":"ok"}' | new DefaultJsonWebhookResponder([msg:"ok"])
     }
 
+    def "test post content types"() {
+        given:
+        controller.rundeckAuthContextProvider = Mock(AuthContextProvider)
+        controller.rundeckAuthContextEvaluator = Mock(AuthContextEvaluator)
+        controller.webhookService = Mock(MockWebhookService)
+
+        when:
+        params.authtoken = "1234"
+        if(isFormData) {
+            params.putAll(payload)
+        } else {
+            request.JSON = payload
+        }
+        request.method = 'POST'
+        request.contentType = contentType
+        controller.post()
+
+        then:
+        1 * controller.webhookService.getWebhookByToken(_) >> { new Webhook(name:"test",authToken: "1234")}
+        1 * controller.rundeckAuthContextProvider.getAuthContextForSubjectAndProject(_, _) >> { new SubjectAuthContext(null, null) }
+        1 * controller.rundeckAuthContextEvaluator.authorizeProjectResourceAny(_,_,_,_) >> { return true }
+        1 * controller.webhookService.processWebhook(_,_,_,_,_) >> { args ->
+            if(isFormData) {
+                args[2].formData["key1"] == "val1"
```

Comment:
this won't fail as a test unless you use `assert`, otherwise since it is within the closure it isn't recognized as a comparison check

Reply:
Thanks, fixed!